src/System.CommandLine/src/System/CommandLine/ArgumentSyntax_Definers.cs

```diff
@@ -53,30 +52,73 @@ public Argument<int> DefineOption(string name, int defaultValue)
             return DefineOption(name, defaultValue, s_int32Parser);
         }
 
-        public Argument<T> DefineOption<T>(string name, ref T value, Func<string, T> valueConverter, string help)
+        public Argument<T> DefineOption<T>(
+            string name,
+            Action<T> assign,
```

Comment:
I think the name `assign` is slightly confusing, I think maybe something like "valueInitializer". Also IMO this should be split into two methods:
- one which takes only assign
- one which takes conditional assign only (i.e. TryParse pattern)